site/src/server.rs

```diff
@@ -615,6 +643,35 @@ where
     }
 }
 
+fn maybe_compressed_response(
+    response: http::response::Builder,
+    body: Vec<u8>,
+    compression: &Option<BrotliEncoderParams>,
+) -> Response {
+    match compression {
+        None => response.body(hyper::Body::from(body)).unwrap(),
+        Some(brotli_params) => {
+            let compressed = compress_bytes(&body, brotli_params);
+            let response = response.header(
+                hyper::header::CONTENT_ENCODING,
+                hyper::header::HeaderValue::from_static("br"),
+            );
+            response.body(hyper::Body::from(compressed)).unwrap()
+        }
+    }
+}
+
+fn compress_bytes(bytes: &Vec<u8>, brotli_params: &BrotliEncoderParams) -> Vec<u8> {
+    let mut compressed = Vec::with_capacity(bytes.len());
+    BrotliCompress(
+        &mut Cursor::new(&bytes),
+        &mut Cursor::new(&mut compressed),
```

Comment:
I would not expect the cursors to be necessary here -- does `BrotliCompress(&bytes[..], &mut compressed, brotli_params)` not work? Can you paste the error message?